### PR TITLE
Query refactor

### PR DESCRIPTION
**Pull request by keshibat**

### CODE REVIEW

File: client/src/Components/Card/BusinessTotalCard.js [hunk 1]
```diff
@@ -1,25 +1,16 @@
-import React, { Component } from "react";
```
Comment (jeznag) [inline client/src/Components/Card/BusinessTotalCard.js]: is this meant to be commented out?

File: client/src/Components/Card/CarNameCard.js [hunk 2]
```diff
@@ -1,27 +1,16 @@
-import React, { Component } from "react";
```
Comment (jeznag) [inline client/src/Components/Card/CarNameCard.js]: likewise

File: client/src/Components/Card/DiagnosticCard.js [hunk 3]
```diff
@@ -1,38 +1,27 @@
...
+
   render() {
     return (
       <Card className="data-cards-r1" >
         <Card.Content>
           <Icon circular inverted color="red" name="wrench" size="big" />
           <div className="data-content">
             <p>Diagnostic Issues    </p>
-            <h2>{this.displayCarData()}</h2>
+            <h2>{this.props.diagnostic}</h2>
```
Comment (jeznag) [inline client/src/Components/Card/DiagnosticCard.js]: prop name is a bit unclear

File: client/src/Components/Card/DiagnosticCard.js [hunk 4]
```diff
@@ -1,38 +1,27 @@
...
-    }
-  }
+
   render() {
     return (
       <Card className="data-cards-r1" >
         <Card.Content>
           <Icon circular inverted color="red" name="wrench" size="big" />
           <div className="data-content">
             <p>Diagnostic Issues    </p>
```
Comment (jeznag) [inline client/src/Components/Card/DiagnosticCard.js]: Unnecessary whitespace

File: client/src/Components/Card/EmissionsCard.js [hunk 5]
```diff
@@ -1,27 +1,17 @@
...
-    }
-  }
+
   render() {
     return (
       <Card className="data-cards-r1" >
         <Card.Content>
           <Icon float="left" circular inverted color="green" name="tree" size="big" />
           <div className="data-content">
             <p>EMISSIONS</p>
```
Comment (jeznag) [inline client/src/Components/Card/EmissionsCard.js]: Inconsistent capitalisation. Other headings aren't capitalised

**Comment by keshibat** [comment]
Dashboard

File: client/src/Components/Card/EmissionsCard.js [hunk 6]
```diff
@@ -32,4 +22,4 @@ class EmissionsCard extends Component {
   }
 }
 
-export default graphql(getEmissionsQuery)(EmissionsCard);
\ No newline at end of file
+export default EmissionsCard;
```
Comment (jeznag) [inline client/src/Components/Card/EmissionsCard.js]: Files should end with a new line. https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files

File: client/src/Components/Card/FuelEconomyCard.js [hunk 7]
```diff
@@ -1,27 +1,17 @@
...
+
   render() {
     return (
       <Card className="data-cards-r1" >
         <Card.Content>
           <Icon circular inverted color="violet" name="dollar sign" size="big" />
           <div className="data-content">
             <p>Fuel Economy</p>
-            <h2>{this.displayCarData()}</h2>
+            <h2>{this.props.fueleconomy}</h2>
```
Comment (jeznag) [inline client/src/Components/Card/FuelEconomyCard.js]: camel case

File: client/src/Components/Card/FuelLeftCard.js [hunk 8]
```diff
@@ -1,41 +1,31 @@
...
-  }
+
   render() {
     return (
-
       <Card className="data-cards-r1" >
         <Card.Content>
           <Icon circular inverted color="orange" name="tachometer alternate" size="big" />
           <div className="data-content">
             <p>Esitmated Fuel Left:</p>
```
Comment (jeznag) [inline client/src/Components/Card/FuelLeftCard.js]: typo `estimated`. Also other cards don't seem to have a colon

File: client/src/Components/Card/FuelLeftCard.js [hunk 9]
```diff
@@ -1,41 +1,31 @@
...
-            <h2>{this.displayCarData()}</h2>
+            <h2>{this.props.fuelleft}</h2>
           </div>
           <Card.Content extra>
             <hr></hr>
-            <p><TraveldSinceCard /></p>
+            <p>
+              {this.props.traveldsince}
```
Comment (jeznag) [inline client/src/Components/Card/FuelLeftCard.js]: typo `traveled`. Camel case

File: client/src/Components/Card/LastFillupCard.js [hunk 10]
```diff
@@ -28,12 +12,12 @@ class LastFillupCard extends Component {
             <Icon float="left" circular inverted color="teal" name="battery high" size="big" />
             <div className="data-content">
               <p>Last Fill Up:</p>
-            <h2>{this.displayCarData()}</h2>
+            <h2>{this.props.fillup}</h2>
```
Comment (jeznag) [inline client/src/Components/Card/LastFillupCard.js]: camel case

File: client/src/Pages/Dashboard/DashboardView.js [hunk 11]
```diff
@@ -1,72 +1,217 @@
...
 
 
 class DashboardView extends Component {
+
+
+
+  displayLastFillupData() {
+    let data = this.props.data;
+    if (data.loading) {
+      return <LoadingSpinner />;
```
Comment (jeznag) [inline client/src/Pages/Dashboard/DashboardView.js]: I wouldn't do this. I'd have one loading spinner for the entire dashboard.

```
render() {
    if (this.props.data.loading) {
      return <LoadingSpinner />;
    }
    return ...
```

File: client/src/Pages/Dashboard/DashboardView.js [hunk 12]
```diff
@@ -1,72 +1,217 @@
...
+    } else {
+      return <div>{data.car.lastfillup}</div>
```
Comment (jeznag) [inline client/src/Pages/Dashboard/DashboardView.js]: Just pass the data not a div

**Comment by jeznag** [comment]
Also:
- you haven't committed package.json
- you should run prettier across the codebase

File: client/src/Components/Card/BusinessTotalCard.js (hunk 1 above)
Comment (keshibat) [reply client/src/Components/Card/BusinessTotalCard.js]: Yes. 
A BusinessTotalCard does not need anymore. That could be inside BusinessRatioCard.
Trying to clean up unnecessary cards.

File: client/src/Components/Card/CarNameCard.js (hunk 2 above)
Comment (keshibat) [reply client/src/Components/Card/CarNameCard.js]: Eventually, I want to display cars based on how many cars the particular user owns. 
Commented out for the time being.

File: client/src/Components/Card/DiagnosticCard.js (hunk 3 above)
Comment (keshibat) [reply client/src/Components/Card/DiagnosticCard.js]: I pushed the latest queryRefactor and client/src/Components/Card/DiagnosticCard.js should be following... 

`    return (
      <Card className="data-cards-r1" >
        <Card.Content>
          <Icon circular inverted color="red" name="wrench" size="big" />
          <div className="data-content">
            <p>Diagnostic Issues    </p>
            <h2>{this.props.diagnostic}</h2>
          </div>
          <Card.Content extra>
            <hr></hr>
            <p>{this.props.diagnosticdetail}</p>

          </Card.Content>
        </Card.Content>
      </Card>
    );`

File: client/src/Components/Card/DiagnosticCard.js (hunk 4 above)
Comment (keshibat) [reply client/src/Components/Card/DiagnosticCard.js]: Yes. Omit white space.

File: client/src/Components/Card/EmissionsCard.js (hunk 5 above)
Comment (keshibat) [reply client/src/Components/Card/EmissionsCard.js]: Fixed as Emissions.

File: client/src/Components/Card/EmissionsCard.js (hunk 6 above)
Comment (keshibat) [reply client/src/Components/Card/EmissionsCard.js]: Thanks for the professional insight. I didn't know.

File: client/src/Components/Card/FuelLeftCard.js (hunk 8 above)
Comment (keshibat) [reply client/src/Components/Card/FuelLeftCard.js]: Fixed typo and omit :

File: client/src/Components/Card/FuelEconomyCard.js (hunk 7 above)
Comment (keshibat) [reply client/src/Components/Card/FuelEconomyCard.js]: fueleconomy => fuelEconomy

File: client/src/Components/Card/LastFillupCard.js (hunk 10 above)
Comment (keshibat) [reply client/src/Components/Card/LastFillupCard.js]: fillup = > fillUp

File: client/src/Pages/Dashboard/DashboardView.js (hunk 12 above)
Comment (keshibat) [reply client/src/Pages/Dashboard/DashboardView.js]: omit a div.

File: client/src/Components/Card/FuelEconomyCard.js (hunk 7 above)
Comment (keshibat) [reply client/src/Components/Card/FuelEconomyCard.js]: fueleconomy => fuelEconomy

File: client/src/Pages/Dashboard/DashboardView.js (hunk 11 above)
Comment (keshibat) [reply client/src/Pages/Dashboard/DashboardView.js]: Make it one loading spinner.
Having a problem with the size of loading spinner which overlaps DashboardSidebar

File: client/src/Components/Card/BusinessRatioCard.js [hunk 13]
```diff
@@ -1,35 +1,23 @@
...
   render() {
     return (
       <Card className="data-cards-r1" >
         <Card.Content>
           <Icon circular inverted color="blue" name="handshake" size="big" />
           <div className="data-content">
             <p>Business Travel</p>
-            <h2>{this.displayCarData()}</h2>
-
+            <h2>{this.props.businessratio}</h2>
```
Comment (jeznag) [inline client/src/Components/Card/BusinessRatioCard.js]: camel case

File: client/src/Pages/Dashboard/DashboardView.js [hunk 14]
```diff
@@ -1,72 +1,144 @@
 import React, { Component } from "react";
-import GoogleMap from "../../Components/GoogleMap/GoogleMap";
-import AntdTableComponent from "../../Components/AntdTable/AntdTableComponent";
+import { graphql } from "react-apollo";
+import { getDashboardAllDataQuery } from "../../queris/queris";
```
Comment (jeznag) [inline client/src/Pages/Dashboard/DashboardView.js]: typo `queries`

File: client/src/Pages/Dashboard/DashboardView.js [hunk 15]
```diff
@@ -1,72 +1,144 @@
...
+import AntdTableComponent from "../../Components/AntdTable/AntdTableComponent";
 
 
 class DashboardView extends Component {
+
+  constructor(props) {
+    super(props)
+  }
+
+  displayLastFillUpData() {
```
Comment (jeznag) [inline client/src/Pages/Dashboard/DashboardView.js]: this methods seem unnecessary. You can directly pass the value as a prop

Comment (keshibat) [reply client/src/Pages/Dashboard/DashboardView.js]: Right.. 
I thought I need a constructor() and super in order to pass the props inside render() method.

File: client/src/Pages/Dashboard/DashboardView.js (hunk 14 above)
Comment (keshibat) [reply client/src/Pages/Dashboard/DashboardView.js]: fixed

File: client/src/Components/Card/BusinessRatioCard.js (hunk 13 above)
Comment (keshibat) [reply client/src/Components/Card/BusinessRatioCard.js]: fixed

File: client/src/Pages/Dashboard/DashboardView.js (hunk 15 above)
Comment (jeznag) [reply client/src/Pages/Dashboard/DashboardView.js]: Not sure what you mean. You can do this:
```
<AverageSpeedCard speed={this.props.data.car.averageSpeed} />
```

There is no value in these methods.

File: client/src/Components/Card/FuelLeftCard.js [hunk 16]
```diff
@@ -1,41 +1,32 @@
...
-            <p>Esitmated Fuel Left:</p>
-            <h2>{this.displayCarData()}</h2>
+            <p>Estimated Fuel Left</p>
+            <h2>{this.props.fuelLeft}</h2>
           </div>
           <Card.Content extra>
             <hr></hr>
-            <p><TraveldSinceCard /></p>
+            <p>
+              {this.props.travelSince}
```
Comment (jeznag) [inline client/src/Components/Card/FuelLeftCard.js]: Should there be a header for this?

Comment (keshibat) [reply client/src/Components/Card/FuelLeftCard.js]: Yes. I will clean up with Sam's Smart Card Component.

File: client/src/Pages/Dashboard/DashboardView.js (hunk 15 above)
Comment (keshibat) [reply client/src/Pages/Dashboard/DashboardView.js]: Right! Cleaned up the unnecessary functions and simply just passing the props.

File: client/.vscode/settings.json [hunk 17]
```diff
@@ -0,0 +1,3 @@
+{
+  "files.insertFinalNewline": true
+}
```
Comment (jeznag) [inline client/.vscode/settings.json]: missing final new line

File: client/src/Components/Card/EmissionsCard.js [hunk 18]
```diff
@@ -1,35 +1,26 @@
...
     return (
       <Card className="data-cards-r1" >
         <Card.Content>
-          <Icon float="left" circular inverted color="green" name="tree" size="big" />
+          <Icon circular inverted color="green" name="tree" size="big" />
           <div className="data-content">
-            <p>EMISSIONS</p>
-            <h2>{this.displayCarData()}</h2>
+            <p>Emissions</p>
+            <h2>{this.props.emission}KG C02</h2>
```
Comment (jeznag) [inline client/src/Components/Card/EmissionsCard.js]: 2 should be subscript. https://www.w3schools.com/TAGS/tag_sub.asp 
http://www.climate-change-wisdom.com/CO2e.html

File: client/src/Components/Card/EmissionsCard.js [hunk 19]
```diff
@@ -1,35 +1,26 @@
...
         </Card.Content>
       </Card>
     );
   }
 }
 
-export default graphql(getEmissionsQuery)(EmissionsCard);
\ No newline at end of file
+export default EmissionsCard;
+
```
Comment (jeznag) [inline client/src/Components/Card/EmissionsCard.js]: Run prettier - shouldn't have these new lines

File: client/src/Components/Card/FuelEconomyCard.js [hunk 20]
```diff
@@ -1,35 +1,28 @@
...
         </Card.Content>
       </Card>
     );
   }
 }
 
-export default graphql(getFuelEconomyQuery)(FuelEconomyCard);
\ No newline at end of file
+export default FuelEconomyCard;
+
```
Comment (jeznag) [inline client/src/Components/Card/FuelEconomyCard.js]: same deal

File: client/src/Components/Card/FuelLeftCard.js [hunk 21]
```diff
@@ -1,41 +1,32 @@
...
   render() {
     return (
-
       <Card className="data-cards-r1" >
         <Card.Content>
           <Icon circular inverted color="orange" name="tachometer alternate" size="big" />
           <div className="data-content">
-            <p>Esitmated Fuel Left:</p>
-            <h2>{this.displayCarData()}</h2>
+            <p>Esitmated Fuel Left</p>
```
Comment (jeznag) [inline client/src/Components/Card/FuelLeftCard.js]: typo

File: client/src/Components/Card/FuelLeftCard.js [hunk 22]
```diff
@@ -1,41 +1,32 @@
...
+            <h2>{this.props.fuelLeft} km</h2>
           </div>
           <Card.Content extra>
             <hr></hr>
-            <p><TraveldSinceCard /></p>
+            <p> {this.props.travelSince}km(since last fill up)</p>
```
Comment (jeznag) [inline client/src/Components/Card/FuelLeftCard.js]: should have more spacing

File: client/src/Components/DashboardSidebar.js [hunk 23]
```diff
@@ -44,31 +44,31 @@ export default class DashboardSidebar extends Component {
             visible={visible}
             width='thin'
           >
-            <Menu.Item as='a'>
-              <Link to="/login">
-                <Icon name='home' />
-                Home
-              </Link>
+            <Menu.Item>
+              WELCOME USER!
```
Comment (jeznag) [inline client/src/Components/DashboardSidebar.js]: Ideally would have the user's name. I'd prefer not to use all caps.

File: client/src/Components/LogoutModal.js [hunk 24]
```diff
@@ -0,0 +1,37 @@
...
+        actions={['No', { positive: true }]}
+        basic
+      >
+
+        <Header icon='log out' content='  Log Out' />
+        <Modal.Content>
+          <p>
+            Are you sure you want to log out?
+      </p>
+        </Modal.Content>
```
Comment (jeznag) [inline client/src/Components/LogoutModal.js]: indentation wrong. Run prettier.

File: client/src/Pages/Dashboard/DashboardView.js [hunk 25]
```diff
@@ -1,72 +1,74 @@
...
-            <EmissionsCard />
-            <FuelEconomyCard />
-          </Card.Group>
-        </div>
-        <div className="ui container">
-          <AntdTableComponent />
-        </div>
-      </>
-    );
+    let data = this.props.data;
```
Comment (jeznag) [inline client/src/Pages/Dashboard/DashboardView.js]: prefer const over let. Convention is to use destructuring. `const { data } = this.props;`

File: client/.vscode/settings.json (hunk 17 above)
Comment (keshibat) [reply client/.vscode/settings.json]: added a final new line.
Is prettier run across all the files including .files?

Comment (jeznag) [reply client/.vscode/settings.json]: It works on .js and .json files. You can customise it to use single quote instead of double quote as well. (I prefer single quote but not a huge deal either way)

File: client/src/Components/Card/EmissionsCard.js (hunk 18 above)
Comment (jeznag) [reply client/src/Components/Card/EmissionsCard.js]: KG should be `kg` http://www.simetricmatters.com/inkorrect.htm

Comment (keshibat) [reply client/src/Components/Card/EmissionsCard.js]: Fixed as follwoing
{this.props.emission}kg C0<sub>2</sub>

File: client/src/Components/Card/EmissionsCard.js (hunk 19 above)
Comment (keshibat) [reply client/src/Components/Card/EmissionsCard.js]: I did run prettier

File: client/src/Components/Card/FuelEconomyCard.js (hunk 20 above)
Comment (keshibat) [reply client/src/Components/Card/FuelEconomyCard.js]: I did run prettier

File: client/src/Components/Card/FuelLeftCard.js (hunk 21 above)
Comment (keshibat) [reply client/src/Components/Card/FuelLeftCard.js]: Fixed typo.

File: client/src/Components/Card/FuelLeftCard.js (hunk 22 above)
Comment (keshibat) [reply client/src/Components/Card/FuelLeftCard.js]: added space in between.

File: client/src/Components/LogoutModal.js (hunk 24 above)
Comment (keshibat) [reply client/src/Components/LogoutModal.js]: I did run prettier.

File: client/src/Pages/Dashboard/DashboardView.js (hunk 25 above)
Comment (keshibat) [reply client/src/Pages/Dashboard/DashboardView.js]: Right. We're not gonna change the value. It should be const.

File: client/src/Components/DashboardSidebar.js (hunk 23 above)
Comment (keshibat) [reply client/src/Components/DashboardSidebar.js]: fixed as Welcome User
